sources/academy/webscraping/web_scraping_for_beginners/crawling/scraping_the_data.md

```diff
@@ -11,221 +11,327 @@ slug: /web-scraping-for-beginners/crawling/scraping-the-data
 
 ---
 
-At the [very beginning of the course](../index.md), we learned that the term web scraping usually means a combined process of data extraction and crawling. And this is exactly what we'll do in this lesson. We will take the code we built in the previous lesson and in the [Basics of data collection](../data_extraction/node_continued.md) section, and we will combine that into a web scraper.
+At the [very beginning of this course](../index.md), we learned that the term web scraping usually means a combined process of data extraction and crawling. And this is exactly what we'll do in this lesson. We will take the crawling code from the previous lesson, and we will combine it with data extraction code and turn everything into a web scraper.
 
-## Quick review of earlier's code {#review-code}
+## Extracting data from a product detail page {#extracting-data}
 
-Here's a quick reminder of earlier's data extraction code:
+The term product detail page (or PDP) is commonly used on e-commerce websites to describe the page where you can find detailed information about a product. In the Warehouse store, it's, for example, [this page describing Denon headphones](https://warehouse-theme-metal.myshopify.com/products/denon-ah-c720-in-ear-headphones).
+
+Let's start writing a script that extracts data from this single PDP. We can use this familiar code as boilerplate.
 
 ```js
-// main.js
+// product.js
```

Comment:
could use the codeblock title here too, for some reason it wont allow me to add suggestion